build.sh

```diff
@@ -1,13 +1,26 @@
 #!/bin/bash
 
-echo $1
-gomod=$(go list -m -json github.com/golangci/golangci-lint@$1 | grep -o '"GoMod": "[^"]*' | grep -o '[^"]*$')
-output=$(cat $gomod)
+set -euo pipefail
+
+YELLOW=$'\e[0;33m'
+NC=$'\e[0m'
+
+[[ -z "${1:-}" ]] && echo "usage: $0 (v1.X.X|latest)" && exit 1
+
+gomod=$(go list -m -json github.com/golangci/golangci-lint@"$1" | grep -o '"GoMod": "[^"]*' | grep -o '[^"]*$')
+output=$(cat "$gomod")
 tools_version=$(grep 'golang.org/x/tools' <<< "$output" | grep -oE '\bv[0-9]+\.[0-9]+\.[0-9]+\b' | grep -oE '[0-9]+\.[0-9]+\.[0-9]+')
+go_version=$(grep -oE '^go [0-9]+\.[0-9].+' <<< "$output" | cut -d ' ' -f2)
+
+local_go_version=$(go version | grep -oE '[0-9]+\.[0-9]+\.[0-9]+')
+[[ "$local_go_version" != "$go_version" ]] && \
+  echo -e "${YELLOW}warning: local go version ($local_go_version) != go version in go.mod from golangci-lint@$1  ($go_version)${NC}" &&
+  echo -e "${YELLOW}continuing build like normal ...${NC}"
 
 go mod init github.com/nilaway-plugin
```

Comment:
**suggestion (non-blocking):**  what do you think if we add here rm go.* before go mod init so incase of the go mod files exists it would be replaced